### DISCUSSION TITLE
Can't setup new Jarvis or Backup after reinstall ioBroker

### DISCUSSION THREAD

**Discussion by Herginator** [Zefau/ioBroker.jarvis#2360] [Q&A]
Hi,
I reinstalled iobroker (via Synology/Docker) from a backup but now I am unable to reinstall Jarvis via backup with error:

Error:
{"errno":-2,"code":"ENOENT","syscall":"open","path":"/opt/iobroker/backups/jarvis_0/states/states.json"}

and setting up a new instance of Jarvis also does not work. When trying to start guild via link an error in browser pops up:
<img width="611" alt="Bildschirm­foto 2023-05-09 um 18 36 40" src="https://github.com/Zefau/ioBroker.jarvis/assets/76481999/fb8180a8-2c46-4a8e-8ac6-f61355c8bbd2">

any ideas?

Best
Robert

**Comment by mcuiobroker** [comment]
Welche Version von backitup genutzt? >2.6.0?
Was hast du genau versucht? Wie?

**Reply by Herginator** [reply]
Ich habe alles aktualisiert, nutze Backitup 2.6.20. Davor hatte ich eine recht alte iobroker mit altem JS Controller und auch altem Node JS. Ich habe vor dem update mit Backitup zunächst eine Sicherung gezogen von iobroker; Jarvis und JavaScripts.

**Comment by mcuiobroker** [comment]
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/46e7c022-6698-41c0-af9f-a462582e0a7e)
Hast du für jarvis eine Extra-Sicherung erhalten?
Und die Rücksicherung klappt dann nicht?
Wie ist denn jetzt der Stand im System Instanzen-Überblick?
`iob diag` laufen lassen und Ergebnisse einstellen -> ioBroker Forum, dann können auch andere besser helfen?

**Reply by Herginator** [reply]
Genau, ich sehe folgende Sicherungen:

<img width="1295" alt="Bildschirmfoto 2023-05-09 um 23 01 15" src="https://github.com/Zefau/ioBroker.jarvis/assets/76481999/0586612a-07f6-4e35-bdc5-dfdd63ec9ac2">

und beim zurückspielen der Jarvis kommt dann am ende der o.g. Fehler.

Instanzen sieht so aus:
<img width="996" alt="Bildschirmfoto 2023-05-09 um 23 04 03" src="https://github.com/Zefau/ioBroker.jarvis/assets/76481999/157cd273-6cdb-4ac7-b9d9-439cff54a7ab">

Wobei die Jarvis nun eine neue Instanz ist, die aber auch nicht per link im browser startet (siehe die andere Fehlermeldung im meinem ersten post).

"iob diag" mache ich wie genau? kann ich das über den Portainer in der Exec Konsole machen?

**Comment by mcuiobroker** [comment]
Stell mal bitte die jarvis-Dateien hier hinein.
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/227ab914-148c-4766-b61b-1dfe9ace0c84)
Sieht extrem klein aus.
Im Prinzip ja, iob diag sollte auf der console laufen.
Aber ich bin nicht der System Experte für alle Systeme, da wäre es mir lieber du würdest es im iobroker-Forum als Thread anlegen.
Thema Jarvis nach Restore nicht lauffähig.
Aber erstmal die Dateien bitte.

**Reply by mcuiobroker** [reply]
ok kannst sie wieder löschen.

**Reply by Herginator** [reply]
und schon mal super lieben Dank, dass Du versuchst, mir zu helfen! :-)

**Comment by Herginator** [comment]
grad nochmal versucht, Jarvis.0 zurückzuspielen. Hier noch der erste Teil:

Started restore ...
[DEBUG] [jarvis] - Start Jarvis Restore ...
[DEBUG] [jarvis] - filename for restore: /opt/iobroker/backups/jarvis.0_2023_05_09-12_44_21_backupiobroker.tar.gz
[DEBUG] [jarvis] - jarvis.0 stopped
[DEBUG] [jarvis] - jarvis tmp directory created: /opt/iobroker/backups/jarvis_0
[DEBUG] [jarvis] - old jarvis database directory was successfully deleted
[EXIT] 0
[DEBUG] [jarvis] - jarvis Restore completed successfully

und nach Button "schließen" kommt dann die Fehlermeldung:
Error:
{"errno":-2,"code":"ENOENT","syscall":"open","path":"/opt/iobroker/backups/jarvis_0/states/states.json"}

**Comment by mcuiobroker** [comment]
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/5192b486-2fed-4d10-ada4-11917b2ae32e)
Die Sicherung soll mit backitup > 2.6.0 gemacht worden sein?
Da fehlen Daten warum auch immer. Die Dateien sind von 2022 und letzte Änderung von 2021.

**Comment by mcuiobroker** [comment]
Dann brauch ich doch die iobroker-Datei

So hätte es aussehen müssen -> mit states-Ordner
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/558c20d1-7f16-4616-8cb5-a0564cf6ddf7)

**Reply by mcuiobroker** [reply]
ja kannst löschen

**Comment by mcuiobroker** [comment]
Welche jarvis Version ist installiert?
Sieht nach Version 3 aus, die Daten sind aber Version v2.2.3?

**Reply by Herginator** [reply]
genau, ist Jarvis 3.1.3

**Reply by mcuiobroker** [reply]
Willst du wieder auf die v2.2.3?
In der v3 hat es einige Veränderungen gegeben. Da dürfte es mehrere Anpassungs-Herausforderungen geben.
Wenn man jetzt die Daten von v2.2.3 einspielen würde, würden diese von der v3.1.3 direkt angepasst werden für die v3.

**Comment by mcuiobroker** [comment]
Herunterladen und Info wenn geschehen, dann nehme ich sie wieder raus.

**Reply by Herginator** [reply]
hab sie runtergeladen

**Reply by mcuiobroker** [reply]
Also vorher entscheiden, was du willst.
Console:
`iob upgrade jarvis@2.2.3 -y `
-> jarvis wird in Version v2.2.3 installiert

**Reply by Herginator** [reply]
ok, und dann nochmal das backup einspielen?

**Reply by Herginator** [reply]
ok, Version ist jetzt wieder 2.2.3. Aber Backup über Backitup bringt am Ende den gleichen Fehler

<img width="876" alt="Bildschirmfoto 2023-05-10 um 00 00 24" src="https://github.com/Zefau/ioBroker.jarvis/assets/76481999/a3ef5c7a-595d-48e6-8474-eeb90c42f877">

**Reply by Herginator** [reply]
btw...was genau mache ich mit den drei Dateien von Dir, die ich runtergeladen habe?

**Reply by mcuiobroker** [reply]
Nicht mit backitup jetzt weitermachen.
Jarvis Instanz stoppen
dann den Inhalt der Dateien in die jarvis.0-DPs einfügen
devices.txt -> jarvis.0.devices
layout.txt -> jarvis.0.layout

**Comment by mcuiobroker** [comment]
Einfügen in die DPs:
- Datei öffnen
- in die Datei klicken -> STRG+a -> alles wird markiert
- STRG+c -> das markierte wir kopiert
- DP auswählen 
- auf den Inhalt dort klicken -> Wert schreiben wird geöffnet
- rein klicken + STRG+a -> alles wird markiert
- STRG+v -> neuer Inhalt wird eingefügt
- Wert setzen klicken

**Reply by Herginator** [reply]
devices konnte ich schreiben, bei Layout allerdings folgende Fehlermeldung:

<img width="650" alt="Bildschirmfoto 2023-05-10 um 00 16 09" src="https://github.com/Zefau/ioBroker.jarvis/assets/76481999/24d67557-00e2-4ca8-9f91-ab652fc61d92">

**Reply by mcuiobroker** [reply]
Wo klickst du hin?
Auf den inhalt klicken nicht den Stift!
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/bca33431-d250-47eb-a9d6-2a5996c2f861)
Weiß nicht was du jetzt gemacht hast, evtl musst du jarvis nochmal löschen und neu anlegen, da du evtl. das Objekt zerstört hats?

**Reply by Herginator** [reply]
mist...ok. Ich installiere jarvis nochmal neu

**Reply by Herginator** [reply]
HAMMER!
Es läuft wieder :-)
Sensationell! Hab super lieben Dank für Deine unglaublich tolle Hilfe!

**Reply by Herginator** [reply]
und das um diese Uhrzeit! Mir fehlen die Worte :-)
Hab eine angenehme Restwoche!
LG
Robert
